src/content/blog/SwitchPackageToProjRef.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Seemless switching between Package and Project references
+date: 2023-11-11
+author: "dawe"
+slug: "2023/11/11"
+---
+
+## Switching seemlessly between Package and Project references - 2023-11-11
+
+##### by dawe
+
+While doing open source development on a code base with package dependencies which are crucial to your work, you easily run into a situation where you want to adjust the package code to your needs. Reasons could be a missing feature, a bug, or just a different opinion on how things should be done.  
+Depending on the nature of your work, you might want to do the changes in the upstream code and consume them right away in the consuming project to finish your original work. This has also the benefit that you see the impact of your changes right away and can adjust them if needed.
+
+We had such a situation recently while working on the [Ionide-Analyzers](https://github.com/ionide/ionide-analyzers). There was a bug in the [FSharp.Analyzers.SDK](https://github.com/ionide/FSharp.Analyzers.SDK) blocking [Florian](https://github.com/nojaf) from finishing his analyzer.
+As our analyzer initiative is still under heavy development, we anticipate to run into this situation more often in the future. So we decided to make it as easy as possible to switch between package and project references to the SDK. Depending on how you consume packages in your project, there are multiple ways to do this. I will show you a convenient way to do this with nuget and MSBuild.
+
+### The setup
+
+We assume you use a [Directory.Packages.props](https://learn.microsoft.com/en-us/nuget/consume-packages/Central-Package-Management) file in your repository to define the package versions you want to use. This file is used by the [Directory.Build.props](https://learn.microsoft.com/en-us/visualstudio/msbuild/customize-by-directory?view=vs-2022) file in the root of your repository to define the package versions for all projects in your repository.  
```

Comment:
I don't think it is super relevant that you mention `Directory.Packages.props`. It wouldn't be the end of the world if the packages file remains as is. The whole point is about no longer referencing them.